xbuild/src/command/build.rs

```diff
@@ -264,6 +264,34 @@ pub fn build(env: &BuildEnv) -> Result<()> {
                     let lib =
                         env.cargo_artefact(&arch_dir.join("cargo"), target, CrateType::Cdylib)?;
                     apk.add_lib(target.android_abi(), &lib)?;
+
+                    let ndk = env.android_ndk();
+
+                    let search_paths = []; // TODO: deps dir and everything from `cargo:rustc-link-search=`
+                    let ndk_sysroot_libs = ndk.join("usr/lib").join(target.ndk_triple());
+                    let provided_libs_paths = [
+                        ndk_sysroot_libs.as_path(),
+                        &*ndk_sysroot_libs.join(
```

Comment:
@dvc94ch Do you remember what was up with this? We can't call `foo.join(..).as_path()` since the temporary owned `PathBuf` created by `fn join()` would be dropped here, but somehow that isn't the case with `&*`??

Reply:
apparently

Reply:
I trust it's safe since the borrow checker isn't complaining, but it still has me bamboozled.

Reply:
https://doc.rust-lang.org/stable/reference/destructors.html#temporary-lifetime-extension Found it, it's relying on scope extension for temporaries. The scope for the results of `join()` are extended to the surrounding scope of the `let` binding, instead of remaining within the expression (`[...]` array constructor) of said `let` binding.

Note that it also/only works with just a `&` on the `PathBuf`, `Deref`ing into `Path`. It doesn't work through `as_path()` since I guess Rust "can't easily follow" what's happening inside that function and as such know that it would have to prolong the lifetime of `self`.

The `*` is just added to force a deref into `Path` first, before turning it into `&Path` again. This can be omitted if the first item is already an `&Path` or the `let` binding would be explicitly annotated to be `&Path` _and not `&PathBuf`.

The more you know :grin: